### PR TITLE
Added accessibility for users with screen readers.

### PR DESCRIPTION
**Pull request by rmcpantoja**
These changes add compatibility and accessibility for users with screen readers via keyboard shortcuts where it will allow you to have easy access to information about compatibility with Windows 11 with these supported screen readers:
▲ JAWS For Windows
▲ Sapi5
## keyboard shortcuts:
CTRL + SHIFT + 1: VIOS type information and if it is compatible.
CTRL + SHIFT + 2: CPU information.
CTRL + SHIFT + 3: GPT information.
CTRL + SHIFT + 4: Installed ram information.
CTRL + SHIFT + 5: Secure boot information.
CTRL + SHIFT + 6: Hard disk information.
CTRL + SHIFT + 7: TPM version information and if it is compatible.
CTRL + SHIFT + c: Copy the last resource of the spoken system to the clipboard where it can be pasted anywhere.

### CODE REVIEW

File: WhyNotWin11.au3
Comment (micwoj92) [inline WhyNotWin11.au3:43]: I think this will need to follow coding style

File: WhyNotWin11.au3
Comment (micwoj92) [inline WhyNotWin11.au3:198]: please remove this and 234, language was removed (was duplicate)

**Review by micwoj92 — COMMENTED** [review]
I also have question about file `nvdaControllerClient32.dll` Could you tell license/origin? Is it possible to redistribute like this? Might be legal blocking issue.

For any real "technical" review you will have to wait till @rcmaehl reviews ;p

**Comment by rmcpantoja** [comment]
Yes, I have the NVDA Controller license, but it is as "license.txt" and in gpl2, however it can replace the one in the program, but I will attach it here anyway.
[license.txt](https://github.com/rcmaehl/WhyNotWin11/files/6736453/license.txt)

**Comment by micwoj92** [comment]
I don't know if these can be combined. Will need checking. Also the CI of your pr is failing for some reason, here log
https://github.com/rcmaehl/WhyNotWin11/pull/258/checks?check_run_id=2946488290#step:6:7

**Review by rmcpantoja — COMMENTED** [review]
Fixed the error of reader.au3

**Comment by htcfreek** [comment]
i suggest to move all third party includes (mvda library, dll, ...) into a subfolder `./include/nvda_library` for better structure.

**Review by htcfreek — COMMENTED** [review]
May be duplicate includes which not necessary and not prevent bby `#include-once`

**Comment by rmcpantoja** [comment]
> i suggest to move all third party includes (mvda library, dll, ...) into a subfolder `./include/nvda_library` for better structure.

It happens that the NVDA DLL needs to be located in the same folder as the main script, because if we include the functions in a subfolder and the NVDA UDF will not work. Likewise, in the compiled version the same must be done.

**Comment by sanny-io** [comment]
Where did you get it from?

**Comment by micwoj92** [comment]
CI still fails, different error this time
```
"D:\a\WhyNotWin11\WhyNotWin11\WhyNotWin11.au3"(963,33) : error: _Translate() called by a previous line with 1 arg(s). Min = 2. First previous line calling this Func is 22.
Func _Translate($iMUI, $sString)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
WhyNotWin11.au3 - 1 error(s), 0 warning(s)
```

https://github.com/rcmaehl/WhyNotWin11/pull/258/checks?check_run_id=2949431532#step:6:8

**Comment by rmcpantoja** [comment]
Hello
This error is not my fault, some change must have been made by @rcmaehl  with the new _translate function. I've already resolved the WhyNotWin11.au3 changes, but now I have to justify the WhyNotWin11_accessibility.au3 to make it translatable.
Or why instead of putting _translate(@osLang, $text) in the original function they put the "Func _Translate($iMUI = @muiLang, $sString)"? It would be a good way to save us the time of putting the @osLang or @muiLang in the first parameter ... Anyway, I'm going to make that change and submit a commit in a few minutes.

**Comment by rcmaehl** [comment]
> Hello
> This error is not my fault, some change must have been made by @rcmaehl with the new _translate function. I've already resolved the WhyNotWin11.au3 changes, but now I have to justify the WhyNotWin11_accessibility.au3 to make it translatable.
> Or why instead of putting _translate(@oslang, $text) in the original function they put the "Func _Translate($iMUI = @muiLang, $sString)"? It would be a good way to save us the time of putting the @oslang or @muiLang in the first parameter ... Anyway, I'm going to make that change and submit a commit in a few minutes.

You can just do @MUILang like it has at the top of the main file. You'll want @MUILang as that is the currently displayed language, which can be different than the language the OS was installed as.

**Comment by rmcpantoja** [comment]
Ready, I have added the @MUILang as the first parameter, since that change seemed to affect the accessibility script.

**Review by rmcpantoja — COMMENTED** [review]
Ready, I have added the @MUILang from _translate function containing a change as the first parameter, since that change seemed to affect the accessibility script.

**Comment by micwoj92** [comment]
I marked it as draft due to that ` nvdaControllerClient32.dll` so it won't be merged by accident. Unless licensing information is 100% clear then we can't merge this. Thanks for understanding.

**Comment by rmcpantoja** [comment]
but could we remove NVDA support for now and just leave sapi5 and JAWS as a solution? I would take care of that.

**Comment by micwoj92** [comment]
> but could we remove NVDA support for now and just leave sapi5 and JAWS as a solution? I would take care of that.

If those 2 solutions do not require extra (binary) files then it is okay. Only NVDA is problematic then.

**Comment by rmcpantoja** [comment]
> > but could we remove NVDA support for now and just leave sapi5 and JAWS as a solution? I would take care of that.
> 
> If those 2 solutions do not require extra (binary) files then it is okay. Only NVDA is problematic then.

Hello, I have deleted the support for NVDA, both the DLL and the UDF, therefore it is now supported to sapi5 and JAWS, so I confirm that these two readers do not need any external DLL to work.

**Comment by micwoj92** [comment]
Now build fails (but CI passes, so I will look into  that)
```
"D:\a\WhyNotWin11\WhyNotWin11\includes\WhyNotWin11_accessibility.au3"(3,10) : error: can't open include file "kbc.au3".
#include "kbc.au3"
~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,28) : warning: $spacebar: possibly used before declaration.
If NOT _ispressed($spacebar)
~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,51) : warning: $up: possibly used before declaration.
If NOT _ispressed($spacebar) or NOT _ispressed($up)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,76) : warning: $down: possibly used before declaration.
If NOT _ispressed($spacebar) or NOT _ispressed($up) or NOT _ispressed($down)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,101) : warning: $left: possibly used before declaration.
If NOT _ispressed($spacebar) or NOT _ispressed($up) or NOT _ispressed($down) or NOT _ispressed($left)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,127) : warning: $right: possibly used before declaration.
If NOT _ispressed($spacebar) or NOT _ispressed($up) or NOT _ispressed($down) or NOT _ispressed($left) or NOT _ispressed($right)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(40,27) : warning: $control: possibly used before declaration.
If not _ispressed($control)
~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(40,46) : warning: $c: possibly used before declaration.
If not _ispressed($control) AND _ispressed($c)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(45,25) : warning: $enter: possibly used before declaration.
If NOT _ispressed($enter)
~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,28) : error: $spacebar: undeclared global variable.
If NOT _ispressed($spacebar)
~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,51) : error: $up: undeclared global variable.
If NOT _ispressed($spacebar) or NOT _ispressed($up)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,76) : error: $down: undeclared global variable.
If NOT _ispressed($spacebar) or NOT _ispressed($up) or NOT _ispressed($down)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,101) : error: $left: undeclared global variable.
If NOT _ispressed($spacebar) or NOT _ispressed($up) or NOT _ispressed($down) or NOT _ispressed($left)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(35,127) : error: $right: undeclared global variable.
If NOT _ispressed($spacebar) or NOT _ispressed($up) or NOT _ispressed($down) or NOT _ispressed($left) or NOT _ispressed($right)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(40,27) : error: $control: undeclared global variable.
If not _ispressed($control)
~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(40,46) : error: $c: undeclared global variable.
If not _ispressed($control) AND _ispressed($c)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
"D:\a\WhyNotWin11\WhyNotWin11\includes\reader.au3"(45,25) : error: $enter: undeclared global variable.
If NOT _ispressed($enter)
~~~~~~~~~~~~~~~~~~~~~~~~^
WhyNotWin11.au3 - 9 error(s), 8 warning(s)
```

**Review by rmcpantoja — COMMENTED** [review]
Reincluding kbc.au3

**Comment by rmcpantoja** [comment]
Is there a way to reverse that commit by restoring the kbc.au3 or do I have to create another pull request? Because I can't find the way to reverse it; I had forgotten that the reader.au3 needs it.

**Comment by micwoj92** [comment]
https://git-scm.com/docs/git-revert

You can also click on you branch on GitHub web, one commit before deletion, do to that file and then copy and reupload

**Comment by rmcpantoja** [comment]
> https://git-scm.com/docs/git-revert
> 
> You can also click on you branch on GitHub web, one commit before deletion, go to that file and then copy and reupload

Ready, the kbc file is available again, so I think there would be no problems when doing the verification and build.
Everything from NVDA has already been permanently removed, now I don't know if there is anything else to merge this pull request.

**Comment by micwoj92** [comment]
> Ready, the kbc file is available again, so I think there would be no problems when doing the verification and build.

Thanks, builds correctly now.

> Everything from NVDA has already been permanently removed, now I don't know if there is anything else to merge this pull request.

Now only final approval from Robert, he's been busy so might take a while.

**Comment by rmcpantoja** [comment]
Thank you very much, but I have a question. Can you remove the draft markup to merge? Since as I understand it you can not. What could we do in that case?
Anyway, I was going to add the documentation in the readme.md about the accessibility script keyboard shortcuts, but this I think I will do when the changes are in the main branch.

**Comment by micwoj92** [comment]
Change to ready for review, still needs Robert's approval. Please have patience, thanks.

**Comment by rmcpantoja** [comment]
Thanks to you, please excuse the inconvenience. I will be pending for anything.

**Comment by htcfreek** [comment]
Regarding nvda dll I found this repo https://github.com/nvaccess/nvda with the original dlls (x64 and x32) and the license.

**Comment by rmcpantoja** [comment]
Indeed, the NVDAControllerClient is on the NVDA Github, and it is the same one where the license under gpl 2.1 had been previously, but if it cannot be included in this type of thing then there would be no problem, although this reader is used by many. So maybe research is being done, and until NVDA is resolved I removed support to just keep JAWS and Sapi5.

**Comment by rcmaehl** [comment]
I'll review what all needs to be done with the license and if possible merge this in with changes. I'm not a fan of using HotKeySet to hijack keys system wide so I will be fixing that, among other readability issues.

**Comment by micwoj92** [comment]
> I'll review what all needs to be done with the license 

That problematic .dll was removed.

**Comment by rmcpantoja** [comment]
Thanks.
Well, the NVDA thing (for now) has been removed because of the licensing issue discussed earlier, but any fixes and help are welcome.

**Comment by micwoj92** [comment]
@rmcpantoja Shows conflicts again, could you resolve when you got time? Thanks.

**Comment by rmcpantoja** [comment]
> @rmcpantoja Shows conflicts again, could you resolve when you got time? Thanks.

Ready, solved.

**Comment by micwoj92** [comment]
@rmcpantoja Another conflict :p

**Comment by rmcpantoja** [comment]
> @rmcpantoja Another conflict :p

Thanks, resolved.

**Review by rcmaehl — APPROVED** [review]

**Pull request merged**
